internal/git/provider/provider.go

```diff
@@ -2,20 +2,79 @@ package provider
 
 import (
 	"fmt"
+	"net/url"
+	"strings"
 
+	"github.com/go-logr/logr"
 	"golang.org/x/net/context"
 )
 
+const GitHubProviderName = "github"
+
 type Provider interface {
-	ListPullRequests(ctx context.Context, repo Repository) []PullRequest
-	AddCommentToPullREquest(ctx context.Context, repo PullRequest, comment []byte)
+	ListPullRequests(ctx context.Context, repo Repository) ([]PullRequest, error)
+	AddCommentToPullRequest(ctx context.Context, repo PullRequest, body []byte) (*Comment, error)
+
+	SetLogger(logr.Logger) error
+	SetToken(tokenType, token string) error
+	SetHostname(hostname string) error
+
+	Setup() error
```

Comment:
If these (Set* and Setup) are intended to be called only by `New` (and perhaps by tests in this package), they can remain unexported. Or put it this way: is it legit to call `New`, then call `Setup` on the result?

Reply:
It is not intended to be call only from `New()`, it is called only from new, but not necessarily.